ah darn, there's a linter error here now too @phisco: https://github.com/crossplane/crossplane/actions/runs/11640854333/job/32418684683?pr=6053

```
+go-lint *failed* | cmd/crank/beta/validate/image.go:71:6: the type name `ErrBaseLayerNotFound` should conform to the `XxxError` format (errname)
+go-lint *failed* | type ErrBaseLayerNotFound struct {
```

--- a/cmd/crank/beta/validate/image.go
+++ b/cmd/crank/beta/validate/image.go
@@ -67,19 +67,25 @@ func (f *Fetcher) FetchImage(image string) ([]conregv1.Layer, error) {
 	return layers, nil
 }
 
-// ErrBaseLayerNotFound is returned when the base layer of the image could not be found.
-type ErrBaseLayerNotFound struct {
-	error
+// BaseLayerNotFoundError is returned when the base layer of the image could not be found.
+type BaseLayerNotFoundError struct {
+	image string
 }
 
-// NewErrBaseLayerNotFound returns a new ErrBaseLayerNotFound error.
-func NewErrBaseLayerNotFound(image string) error {
-	return &ErrBaseLayerNotFound{errors.Errorf("no base layer found for image %s", image)}
+// Error implements the error interface.
+func (e *BaseLayerNotFoundError) Error() string {
+	return fmt.Sprintf("base layer not found for the image %s", e.image)
 }
 
-// IsErrBaseLayerNotFound checks if the error is of type ErrBaseLayerNotFound.
+// NewBaseLayerNotFoundError returns a new BaseLayerNotFoundError error.
+func NewBaseLayerNotFoundError(image string) error {
+	return &BaseLayerNotFoundError{image: image}
+}
+
+// IsErrBaseLayerNotFound checks if the error is of type BaseLayerNotFoundError.
 func IsErrBaseLayerNotFound(err error) bool {
-	return errors.Is(err, &ErrBaseLayerNotFound{})
+	var e *BaseLayerNotFoundError
+	return errors.As(err, &e)
 }
 
 // FetchBaseLayer fetches the base layer of the image which contains the 'package.yaml' file.
@@ -112,7 +118,7 @@ func (f *Fetcher) FetchBaseLayer(image string) (*conregv1.Layer, error) {
 		}
 	}
 	if label == "" {
-		return nil, NewErrBaseLayerNotFound(image)
+		return nil, NewBaseLayerNotFoundError(image)
 	}
 
 	lDigest := strings.SplitN(label, ":", 2)[1] // e.g.: sha256:0158764f65dc2a68728fdffa6ee6f2c9ef158f2dfed35abbd4f5bef8973e4b59

--- a/cmd/crank/beta/validate/image_test.go
+++ b/cmd/crank/beta/validate/image_test.go
@@ -111,3 +111,36 @@ func TestFindImageTagForVersionConstraint(t *testing.T) {
 		})
 	}
 }
+
+func TestIsErrBaseLayerNotFound(t *testing.T) {
+	type args struct {
+		err error
+	}
+	tests := map[string]struct {
+		reason string
+		args   args
+		want   bool
+	}{
+		"BaseLayerNotFound": {
+			reason: "Should return true if the error is a BaseLayerNotFound error",
+			args: args{
+				err: NewBaseLayerNotFoundError("foo"),
+			},
+			want: true,
+		},
+		"NotBaseLayerNotFound": {
+			reason: "Should return false if the error is not a BaseLayerNotFound error",
+			args: args{
+				err: fmt.Errorf("foo"),
+			},
+			want: false,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			if got := IsErrBaseLayerNotFound(tt.args.err); got != tt.want {
+				t.Errorf("IsErrBaseLayerNotFound() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}

